cli: panic in doctor causes failure in acceptance/version-upgrade

**Describe the problem**

Panic on

https://github.com/cockroachdb/cockroach/blob/f501f170bf01b476b6170ef472c3b4e3993b0a16/pkg/cli/doctor.go#L226

via debug zip.

```
panic: interface conversion: driver.Value is nil, not []uint8 [recovered]
        panic: interface conversion: driver.Value is nil, not []uint8

goroutine 1 [running]:
panic(0x42d9320, 0xc0002f0480)
        /usr/local/go/src/runtime/panic.go:1064 +0x545 fp=0xc0013b5628 sp=0xc0013b5560 pc=0x4e1825
github.com/cockroachdb/cockroach/pkg/util/log/logcrash.RecoverAndReportPanic(0x5643ae0, 0xc000130058, 0xc000668000)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/logcrash/crash_reporting.go:90 +0x9f fp=0xc0013b5678 sp=0xc0013b5628 pc=0x14b04df
runtime.call32(0x0, 0x4ff7488, 0xc001308e88, 0x1800000018)
        /usr/local/go/src/runtime/asm_amd64.s:540 +0x3e fp=0xc0013b56a8 sp=0xc0013b5678 pc=0x5191de
runtime.reflectcallSave(0xc0013b57e8, 0x4ff7488, 0xc001308e88, 0x18)
        /usr/local/go/src/runtime/panic.go:881 +0x58 fp=0xc0013b56d8 sp=0xc0013b56a8 pc=0x4e1218
runtime.runOpenDeferFrame(0xc000000180, 0xc001308e40, 0x0)
        /usr/local/go/src/runtime/panic.go:855 +0x2cd fp=0xc0013b5768 sp=0xc0013b56d8 pc=0x4e10cd
panic(0x42d9320, 0xc0002f0480)
        /usr/local/go/src/runtime/panic.go:969 +0x1b9 fp=0xc0013b5830 sp=0xc0013b5768 pc=0x4e1499
runtime.panicdottypeE(0x0, 0x4116aa0, 0x4296fe0)
        /usr/local/go/src/runtime/iface.go:261 +0xaf fp=0xc0013b5850 sp=0xc0013b5830 pc=0x4b52af
github.com/cockroachdb/cockroach/pkg/cli.runClusterDoctor.func3(0xc000bcee00, 0x4, 0x4, 0x4, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/doctor.go:226 +0x325 fp=0xc0013b58d0 sp=0xc0013b5850 pc=0x3b12f25
github.com/cockroachdb/cockroach/pkg/cli.selectRowsMap(0xc000c26320, 0x49fe37f, 0x35, 0xc000bcee00, 0x4, 0x4, 0xc0013b5a08, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/doctor.go:390 +0x118 fp=0xc0013b5968 sp=0xc0013b58d0 pc=0x3ac5978
github.com/cockroachdb/cockroach/pkg/cli.runClusterDoctor(0x0, 0x0, 0x0, 0x0, 0xc000c26320, 0x55a6980, 0xc000c02570, 0x2540be400, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/doctor.go:217 +0x46b fp=0xc0013b5aa0 sp=0xc0013b5968 pc=0x3ac4ceb
github.com/cockroachdb/cockroach/pkg/cli.runDebugZip(0x77a1220, 0xc0014f8480, 0x1, 0x3, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/zip.go:348 +0xd99 fp=0xc0013b7ac8 sp=0xc0013b5aa0 pc=0x3b02559
```

**To Reproduce**

Run acceptance/version-upgrade a few times.

**Expected behavior**
Either a more descriptive error, or success. (It's well possible that the underlying cluster wasn't health).

**Additional data / screenshots**

**Environment:**
master at time of writing

**Additional context**

--- a/pkg/cli/doctor.go
+++ b/pkg/cli/doctor.go
@@ -223,7 +223,12 @@ FROM system.namespace`
 			return err
 		}
 		md.Progress = &jobspb.Progress{}
-		if err := protoutil.Unmarshal(vals[3].([]byte), md.Progress); err != nil {
+		// Progress is a nullable column, so have to check for nil here.
+		progressBytes, ok := vals[3].([]byte)
+		if !ok {
+			return errors.Errorf("unexpected NULL progress on job row: %v", md)
+		}
+		if err := protoutil.Unmarshal(progressBytes, md.Progress); err != nil {
 			return err
 		}
 		if md.Status == jobs.StatusRunning {

